### ISSUE TITLE
[APP-840] Translate link failing in some cases

### ISSUE THREAD

**Issue by pfrazee** [bluesky-social/social-app#1312]
Some Android users have been reporting that the translate link fails in some cases. I'm unsure if this is due to the presence of specific characters or if the link is just busted on android, but we should check on both.

<sub>[APP-840](https://linear.app/blueskyweb/issue/APP-840/translate-link-failing-in-some-cases)</sub>

**Comment by estrattonbailey** [comment]
Hey @maisondasilva 👋 thanks for chiming in.

Do you happen to have one or two examples of posts that fail to translate for you? Would be super helpful 🙏 thanks!

**Comment by maisondasilva** [comment]
@estrattonbailey for me any post does not work. I will upload a video. 

https://bsky.app/profile/bsky.app/post/3k62vnbg6oh23

**Comment by Zero3K** [comment]
I am not getting a translate link. Here's a screenshot:

![image](https://github.com/bluesky-social/social-app/assets/6194332/96eb45d0-40c6-4b23-9311-c72f19ede726)

**Comment by maisondasilva** [comment]
> 

see if you go to settings then language and then select only your language again appears

**Comment by Zero3K** [comment]
I see it. Both the language selector and translate link are broken. Selecting English in the selector doesn't hide the Japanese posts in my Following feed and clicking on translate doesn't default to translating from the automatically detected language to English and the left side isn't filled in with the Japanese text.

**Comment by maisondasilva** [comment]
> I see it. Both the language selector and translate link are broken. Selecting English in the selector doesn't hide the Japanese posts in my Following feed and clicking on translate doesn't default to translating from the automatically detected language to English and the left side isn't filled in with the Japanese text. Also, the Translate link should show up even if I choose to view all posts no matter the language.

I have selected in Portuguese and I see content in English, I think that part is not 100%

**Comment by estrattonbailey** [comment]
Thanks for the additional info 👍 opened a PR to address part of this.

>I have selected in Portuguese and I see content in English

That's correct, although it might be confusing. In your "Following" feed, these are people you follow, so we don't filter out any posts in other languages. However, if you have a language selected in your "Content Languages", you _should_ see a translate link.

_We do_ filter out posts in other languages in "custom" feeds aka feeds other than your "Following" feed.

**Comment by Zero3K** [comment]
That makes sense.

**Comment by estrattonbailey** [comment]
Bringing the convo back here.

As we discovered with @maisondasilva on [the PR](https://github.com/bluesky-social/social-app/pull/1345#issuecomment-1714433536) (many thanks 🙌 ), translation links are still failing if the user has the Google Translate app installed.

This worked previously because we specified a language to translate into via the `tl=<lang>` query param. While the web interface supports `tl=auto`, the Android Translate app deeplink does not. It must be a valid language specifier such as `en`.

So this is maybe a small bug in the Google Translate app that may be fixed upstream eventually. But in the mean time, we have some work on languages coming up where we'd like to set the user's preferred "primary language" separately from "content languages", and that preference could provide a value for the translate link so that it works with the Android app. I'll leave this open until we get that update in in the coming weeks.
